src/main/java/org/cryptomator/ui/mainwindow/ResizeController.java

```diff
@@ -1,41 +1,97 @@
 	public Region brResizer;
+	public Region tResizer;
+	public Region rResizer;
+	public Region bResizer;
+	public Region lResizer;
 
 	private double origX, origY, origW, origH;
 
+	private final Settings settings;
+
+	private final BooleanBinding showResizingArrows;
+
 	@Inject
-	ResizeController(@MainWindow Stage window) {
+	ResizeController(@MainWindow Stage window, Settings settings) {
 		this.window = window;
-		// TODO inject settings and save current position and size
+		this.settings = settings;
+		this.showResizingArrows = Bindings.createBooleanBinding(this::isShowResizingArrows, window.fullScreenProperty());
 	}
 
 	@FXML
 	public void initialize() {
-		tlResizer.setOnMousePressed(this::startResize);
-		trResizer.setOnMousePressed(this::startResize);
-		blResizer.setOnMousePressed(this::startResize);
-		brResizer.setOnMousePressed(this::startResize);
-		tlResizer.setOnMouseDragged(this::resizeTopLeft);
-		trResizer.setOnMouseDragged(this::resizeTopRight);
-		blResizer.setOnMouseDragged(this::resizeBottomLeft);
-		brResizer.setOnMouseDragged(this::resizeBottomRight);
+		LOG.trace("init ResizeController");
+
+		if (neverTouched()){
+			settings.displayConfigurationProperty().setValue(getMonitorSizes());
+			return;
+		} else {
+			if (checkForChangedDisplayConfiguration()) {
+				//If the position is illegal, then the window appears on the main screen in the middle of the window.
+				Dimension size = Toolkit.getDefaultToolkit().getScreenSize();
+				window.setX((size.getWidth() - window.getMinWidth()) / 2);
+				window.setY((size.getHeight() - window.getMinHeight()) / 2);
+				window.setWidth(window.getMinWidth());
+				window.setHeight(window.getMinHeight());
+			}
+			else {
+				window.setHeight(settings.windowHeightProperty().get() > window.getMinHeight() ? settings.windowHeightProperty().get() : window.getMinHeight());
+				window.setWidth(settings.windowWidthProperty().get() > window.getMinWidth() ? settings.windowWidthProperty().get() : window.getMinWidth());
+				window.setX(settings.windowXPositionProperty().get());
+				window.setY(settings.windowYPositionProperty().get());
+			}
+		}
+		savePositionalSettings();
+	}
+
+	private boolean neverTouched(){
+		return (settings.windowHeightProperty().get() == 0) && (settings.windowWidthProperty().get() == 0) &&
+				(settings.windowXPositionProperty().get() == 0) && (settings.windowYPositionProperty().get() == 0);
+	}
+
+	private boolean checkForChangedDisplayConfiguration(){
+		String currentDisplayConfiguration = getMonitorSizes();
+		String settingsDisplayConfiguration = settings.displayConfigurationProperty().get();
+		boolean changedConfiguration = !settingsDisplayConfiguration.equals(currentDisplayConfiguration);
+		if (changedConfiguration) settings.displayConfigurationProperty().setValue(currentDisplayConfiguration);
+		return changedConfiguration;
+	}
+
+	private String getMonitorSizes() {
+		GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
```

Comment:
Use the javafx `Stage::getScreens` instead of the AWT classes and operate on the primary screen. (see https://openjfx.io/javadoc/16/javafx.graphics/javafx/stage/Screen.html)